workshops/models.py

```diff
@@ -397,6 +400,42 @@ def is_staff(self):
     def is_admin(self):
         return is_admin(self)
 
+    @property
+    def missing_swc_instructor_requirements(self):
+        """ Returns set of requirements that are not passed yet by the
+        trainee and are mandatory to become SWC Instructor.. """
+        training = TrainingRequirement.objects.get(name='Training')
+        homework = TrainingRequirement.objects.get(name='SWC Homework')
+        discussion = TrainingRequirement.objects.get(name='Discussion')
+        demo = TrainingRequirement.objects.get(name='SWC Demo')
+
+        requirements = {training, homework, discussion, demo}
+        progresses = self.trainingprogress_set.filter(discarded=False, state='p')
+        passed_requirements = {p.requirement for p in progresses}
+        return requirements - passed_requirements
```

Comment:
Sth like this does only one query per person:
```python
    @property
    def missing_swc_instructor_requirements(self):
        """Returns set of requirements' names (list of strings) that are not
        passed yet by the trainee and are mandatory to become SWC Instructor.
        """
        requirements = [
            'Training',
            'SWC Homework',
            'Discussion',
            'SWC Demo'
        ]
        passed = self.trainingprogress_set\
                     .filter(discarded=False, state='p',
                             requirement__name__in=requirements) \
                     .values_list('requirement__name', flat=True)
        return set(requirements) - set(passed)
```
Unfortunately, this is used in a view for >1 person and I have no idea how to make it faster.